File: truevfs-kernel/truevfs-kernel-impl/src/main/scala/net/truevfs/kernel/impl/FalsePositiveArchiveException.scala
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.truevfs.kernel.impl

import net.java.truecommons.shed._
import java.io._
import javax.annotation.concurrent._

/** Indicates that a file system is a false positive file system.
  * 
  * This exception type is reserved for non-local control flow in
  * file system controller chains in order to reroute file system operations to
  * the parent file system of a false positive federated (archive) file system.
  *
  * @see    FalsePositiveArchiveController
  * @author Christian Schlichtherle
  */
@Immutable
private class FalsePositiveArchiveException(cause: IOException)
extends ControlFlowException(cause) {
  override def getCause() = super.getCause.asInstanceOf[IOException]
}

File: truevfs-kernel/truevfs-kernel-impl/src/main/java/net/java/truevfs/kernel/impl/FalsePositiveArchiveException.java
/*
 * Copyright © 2005 - 2021 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.kernel.impl;

import net.java.truecommons.shed.ControlFlowException;

import java.io.IOException;

/**
 * Indicates that a file system is a false positive file system.
 * <p>
 * This exception type is reserved for non-local control flow in file system controller chains in order to reroute file
 * system operations to the parent file system of a false positive federated (archive) file system.
 *
 * @author Christian Schlichtherle
 * @see FalsePositiveArchiveController
 */
class FalsePositiveArchiveException extends ControlFlowException {

    FalsePositiveArchiveException(IOException cause) {
        super(cause);
    }

    @Override
    public IOException getCause() {
        return (IOException) super.getCause();
    }
}
